[Improvement] Column.java Prefer instanceof to getClass

### What would you like to be improved?

Column.java:180: warning: [EqualsGetClass] Prefer instanceof to getClass when implementing Object#equals.
    public boolean equals(Object o) {

See https://errorprone.info/bugpattern/EqualsGetClass.

### How should we improve?

_No response_

--- a/api/src/main/java/com/datastrato/gravitino/rel/Column.java
+++ b/api/src/main/java/com/datastrato/gravitino/rel/Column.java
@@ -181,7 +181,7 @@ public interface Column {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof ColumnImpl)) {
         return false;
       }
       ColumnImpl column = (ColumnImpl) o;

--- a/api/src/main/java/com/datastrato/gravitino/rel/expressions/transforms/Transform.java
+++ b/api/src/main/java/com/datastrato/gravitino/rel/expressions/transforms/Transform.java
@@ -74,7 +74,7 @@ public interface Transform extends Expression {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof SingleFieldTransform)) {
         return false;
       }
       SingleFieldTransform that = (SingleFieldTransform) o;

--- a/api/src/main/java/com/datastrato/gravitino/rel/partitions/Partitions.java
+++ b/api/src/main/java/com/datastrato/gravitino/rel/partitions/Partitions.java
@@ -108,7 +108,7 @@ public class Partitions {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof RangePartitionImpl)) {
         return false;
       }
       RangePartitionImpl that = (RangePartitionImpl) o;
@@ -158,7 +158,7 @@ public class Partitions {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof ListPartitionImpl)) {
         return false;
       }
       ListPartitionImpl that = (ListPartitionImpl) o;
@@ -217,7 +217,7 @@ public class Partitions {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof IdentityPartitionImpl)) {
         return false;
       }
       IdentityPartitionImpl that = (IdentityPartitionImpl) o;

--- a/api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java
+++ b/api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java
@@ -110,7 +110,7 @@ public class Decimal {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof Decimal)) {
       return false;
     }
     Decimal decimal = (Decimal) o;

--- a/api/src/main/java/com/datastrato/gravitino/rel/types/Types.java
+++ b/api/src/main/java/com/datastrato/gravitino/rel/types/Types.java
@@ -243,7 +243,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof DecimalType)) {
         return false;
       }
       DecimalType that = (DecimalType) o;
@@ -470,7 +470,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof FixedType)) {
         return false;
       }
       FixedType fixedType = (FixedType) o;
@@ -520,7 +520,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof VarCharType)) {
         return false;
       }
       VarCharType that = (VarCharType) o;
@@ -570,7 +570,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof FixedCharType)) {
         return false;
       }
       FixedCharType that = (FixedCharType) o;
@@ -649,7 +649,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof StructType)) {
         return false;
       }
       StructType that = (StructType) o;
@@ -752,7 +752,7 @@ public class Types {
         if (this == o) {
           return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (!(o instanceof Field)) {
           return false;
         }
         Field field = (Field) o;
@@ -849,7 +849,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof ListType)) {
         return false;
       }
       ListType listType = (ListType) o;
@@ -945,7 +945,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof MapType)) {
         return false;
       }
       MapType mapType = (MapType) o;
@@ -1004,7 +1004,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof UnionType)) {
         return false;
       }
       UnionType unionType = (UnionType) o;
@@ -1059,7 +1059,7 @@ public class Types {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof UnparsedType)) {
         return false;
       }
       UnparsedType that = (UnparsedType) o;

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -295,7 +295,8 @@ subprojects {
         "UnusedAnonymousClass",
         "UnusedCollectionModifiedInPlace",
         "VarTypeName",
-        "XorPower"
+        "XorPower",
+        "EqualsGetClass"
       )
     }
   }

--- a/catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/bean/JdbcIndexBean.java
+++ b/catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/bean/JdbcIndexBean.java
@@ -45,7 +45,7 @@ public class JdbcIndexBean {
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
+    if (!(o instanceof JdbcIndexBean)) return false;
     JdbcIndexBean that = (JdbcIndexBean) o;
     return order == that.order
         && indexType == that.indexType

--- a/catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java
+++ b/catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/converter/JdbcTypeConverter.java
@@ -73,7 +73,7 @@ public abstract class JdbcTypeConverter<TO> {
     @Override
     public boolean equals(Object o) {
       if (this == o) return true;
-      if (o == null || getClass() != o.getClass()) return false;
+      if (!(o instanceof JdbcTypeBean)) return false;
       JdbcTypeBean typeBean = (JdbcTypeBean) o;
       return Objects.equals(typeName, typeBean.typeName)
           && Objects.equals(columnSize, typeBean.columnSize)

--- a/clients/client-java/src/test/java/com/datastrato/gravitino/client/TestHTTPClient.java
+++ b/clients/client-java/src/test/java/com/datastrato/gravitino/client/TestHTTPClient.java
@@ -270,7 +270,7 @@ public class TestHTTPClient {
       if (this == o) {
         return true;
       }
-      if (o == null || getClass() != o.getClass()) {
+      if (!(o instanceof Item)) {
         return false;
       }
       Item item = (Item) o;

--- a/common/src/main/java/com/datastrato/gravitino/dto/MetalakeDTO.java
+++ b/common/src/main/java/com/datastrato/gravitino/dto/MetalakeDTO.java
@@ -158,7 +158,7 @@ public class MetalakeDTO implements Metalake {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof MetalakeDTO)) {
       return false;
     }
     MetalakeDTO that = (MetalakeDTO) o;

--- a/common/src/main/java/com/datastrato/gravitino/dto/rel/DistributionDTO.java
+++ b/common/src/main/java/com/datastrato/gravitino/dto/rel/DistributionDTO.java
@@ -159,7 +159,7 @@ public class DistributionDTO implements Distribution {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof DistributionDTO)) {
       return false;
     }
 

--- a/common/src/main/java/com/datastrato/gravitino/dto/rel/indexes/IndexDTO.java
+++ b/common/src/main/java/com/datastrato/gravitino/dto/rel/indexes/IndexDTO.java
@@ -61,7 +61,7 @@ public class IndexDTO implements Index {
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
+    if (!(o instanceof IndexDTO)) return false;
     IndexDTO indexDTO = (IndexDTO) o;
 
     return indexType == indexDTO.indexType

--- a/common/src/main/java/com/datastrato/gravitino/dto/rel/partitions/IdentityPartitionDTO.java
+++ b/common/src/main/java/com/datastrato/gravitino/dto/rel/partitions/IdentityPartitionDTO.java
@@ -131,7 +131,7 @@ public class IdentityPartitionDTO implements PartitionDTO, IdentityPartition {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof IdentityPartitionDTO)) {
       return false;
     }
     IdentityPartitionDTO that = (IdentityPartitionDTO) o;

--- a/common/src/main/java/com/datastrato/gravitino/dto/rel/partitions/ListPartitionDTO.java
+++ b/common/src/main/java/com/datastrato/gravitino/dto/rel/partitions/ListPartitionDTO.java
@@ -112,7 +112,7 @@ public class ListPartitionDTO implements PartitionDTO, ListPartition {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof ListPartitionDTO)) {
       return false;
     }
     ListPartitionDTO that = (ListPartitionDTO) o;

--- a/common/src/main/java/com/datastrato/gravitino/dto/rel/partitions/RangePartitionDTO.java
+++ b/common/src/main/java/com/datastrato/gravitino/dto/rel/partitions/RangePartitionDTO.java
@@ -63,7 +63,7 @@ public class RangePartitionDTO implements PartitionDTO, RangePartition {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof RangePartitionDTO)) {
       return false;
     }
     RangePartitionDTO that = (RangePartitionDTO) o;

--- a/core/src/main/java/com/datastrato/gravitino/lock/TreeLockNode.java
+++ b/core/src/main/java/com/datastrato/gravitino/lock/TreeLockNode.java
@@ -191,7 +191,7 @@ public class TreeLockNode {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof TreeLockNode)) {
       return false;
     }
     TreeLockNode that = (TreeLockNode) o;

--- a/core/src/main/java/com/datastrato/gravitino/meta/CatalogEntity.java
+++ b/core/src/main/java/com/datastrato/gravitino/meta/CatalogEntity.java
@@ -238,7 +238,7 @@ public class CatalogEntity implements Entity, Auditable, HasIdentifier {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof CatalogEntity)) {
       return false;
     }
     CatalogEntity that = (CatalogEntity) o;

--- a/core/src/main/java/com/datastrato/gravitino/meta/SchemaEntity.java
+++ b/core/src/main/java/com/datastrato/gravitino/meta/SchemaEntity.java
@@ -133,7 +133,7 @@ public class SchemaEntity implements Entity, Auditable, HasIdentifier {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof SchemaEntity)) {
       return false;
     }
     SchemaEntity schema = (SchemaEntity) o;

--- a/core/src/main/java/com/datastrato/gravitino/meta/TableEntity.java
+++ b/core/src/main/java/com/datastrato/gravitino/meta/TableEntity.java
@@ -101,7 +101,7 @@ public class TableEntity implements Entity, Auditable, HasIdentifier {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof TableEntity)) {
       return false;
     }
 

--- a/trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/SortingColumn.java
+++ b/trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/SortingColumn.java
@@ -82,7 +82,7 @@ public class SortingColumn {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof SortingColumn)) {
       return false;
     }
 

